vm/src/stdlib/stat.rs

```diff
@@ -13,208 +10,229 @@ mod stat {
 
-    #[cfg(unix)]
     #[pyattr]
-    pub const S_IRUSR: Mode = libc::S_IRUSR;
-    #[cfg(not(unix))]
+    pub const S_IRWXG: Mode = unix_libc_get!(S_IRWXG, 0o0070);
+
     #[pyattr]
-    pub const S_IRUSR: Mode = 0o0400;
+    pub const S_IRGRP: Mode = unix_libc_get!(S_IRGRP, 0o0040);
 
-    #[cfg(unix)]
     #[pyattr]
-    pub const S_IREAD: Mode = libc::S_IRUSR;
-    #[cfg(not(unix))]
+    pub const S_IWGRP: Mode = unix_libc_get!(S_IWGRP, 0o0020);
+
     #[pyattr]
-    pub const S_IREAD: Mode = 0o0400;
+    pub const S_IXGRP: Mode = unix_libc_get!(S_IXGRP, 0o0010);
 
-    #[cfg(unix)]
     #[pyattr]
-    pub const S_IWUSR: Mode = libc::S_IWUSR;
-    #[cfg(not(unix))]
+    pub const S_IRWXO: Mode = unix_libc_get!(S_IRWXO, 0o0007);
+
     #[pyattr]
-    pub const S_IWUSR: Mode = 0o0200;
+    pub const S_IROTH: Mode = unix_libc_get!(S_IROTH, 0o0004);
 
-    #[cfg(all(unix, not(target_os = "android"), not(target_os = "redox")))]
     #[pyattr]
-    pub const S_IWRITE: Mode = libc::S_IWRITE;
-    #[cfg(any(not(unix), target_os = "android", target_os = "redox"))]
+    pub const S_IWOTH: Mode = unix_libc_get!(S_IWOTH, 0o0002);
+
     #[pyattr]
-    pub const S_IWRITE: Mode = 0o0200;
+    pub const S_IXOTH: Mode = unix_libc_get!(S_IXOTH, 0o0001);
 
-    #[cfg(unix)]
     #[pyattr]
-    pub const S_IXUSR: Mode = libc::S_IXUSR;
-    #[cfg(not(unix))]
+    pub const S_IWRITE: Mode = {
+        cfg_if::cfg_if! {
+            if #[cfg(all(unix, not(target_os = "android"), not(target_os = "redox")))] {
+                libc::S_IWRITE
+            } else {
+                0o0200
+            }
+        }
+    };
+
     #[pyattr]
-    pub const S_IXUSR: Mode = 0o0100;
+    pub const S_IEXEC: Mode = {
+        cfg_if::cfg_if! {
+            if #[cfg(all(unix, not(target_os = "android"), not(target_os = "redox")))] {
+                libc::S_IEXEC
+            } else {
+                0o0100
+            }
+        }
+    };
+
+    // Windows file attributes (if on Windows)
 
-    #[cfg(all(unix, not(target_os = "android"), not(target_os = "redox")))]
+    #[cfg(windows)]
     #[pyattr]
-    pub const S_IEXEC: Mode = libc::S_IEXEC;
-    #[cfg(any(not(unix), target_os = "android", target_os = "redox"))]
+    pub use windows_sys::Win32::Storage::FileSystem::{
+        FILE_ATTRIBUTE_ARCHIVE, FILE_ATTRIBUTE_COMPRESSED, FILE_ATTRIBUTE_DEVICE,
+        FILE_ATTRIBUTE_DIRECTORY, FILE_ATTRIBUTE_ENCRYPTED, FILE_ATTRIBUTE_HIDDEN,
+        FILE_ATTRIBUTE_INTEGRITY_STREAM, FILE_ATTRIBUTE_NO_SCRUB_DATA, FILE_ATTRIBUTE_NORMAL,
+        FILE_ATTRIBUTE_NOT_CONTENT_INDEXED, FILE_ATTRIBUTE_OFFLINE, FILE_ATTRIBUTE_READONLY,
+        FILE_ATTRIBUTE_REPARSE_POINT, FILE_ATTRIBUTE_SPARSE_FILE, FILE_ATTRIBUTE_SYSTEM,
+        FILE_ATTRIBUTE_TEMPORARY, FILE_ATTRIBUTE_VIRTUAL,
+    };
+
+    // Unix file flags (if on Unix)
+
     #[pyattr]
-    pub const S_IEXEC: Mode = 0o0100;
+    pub const UF_NODUMP: u32 = macos_libc_get!(UF_NODUMP, 0x00000001);
```

Comment:
Then this one also will have the same pattern
```suggestion
    pub const UF_NODUMP: u32 = libc_const!(#[cfg(target_os = "macos")] UF_NODUMP, 0x00000001);
```